src/main/java/org/broadinstitute/consent/http/models/support/TicketFactory.java

```diff
@@ -0,0 +1,84 @@
+package org.broadinstitute.consent.http.models.support;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import java.util.List;
+import org.broadinstitute.consent.http.enumeration.SupportRequestType;
+import org.broadinstitute.consent.http.util.gson.GsonUtil;
+import org.zendesk.client.v2.model.Comment;
+import org.zendesk.client.v2.model.CustomFieldValue;
+import org.zendesk.client.v2.model.Request;
+import org.zendesk.client.v2.model.Ticket;
+
+public class TicketFactory {
+
+  /**
+   * Generate the created Request object from the Zendesk request response content.
+   *
+   * @param response The response content from the Zendesk Request API
+   * @return Parsed request.
+   */
+  public Request parseRequestResponse(String response) {
+    Gson gson = GsonUtil.getInstance();
+    JsonObject obj = gson.fromJson(response, JsonObject.class);
+    JsonObject request = obj.get("request").getAsJsonObject();
+    return gson.fromJson(request, Request.class);
+  }
+
+  /**
+   * Constructs a DuosTicket with the proper structure to request support via Zendesk
+   *
+   * @param name        The name of the user requesting support
+   * @param type        The type of request ("question", "incident", "problem", "task")
+   * @param email       The email of the user requesting support
+   * @param subject     Subject line of the request
+   * @param description Description of the task or question
+   * @param url         The origin url of this request
+   * @param uploads     Optional list of attachment tokens
+   */
+  public DuosTicket createTicket(String name, SupportRequestType type, String email, String subject,
+      String description, String url, List<String> uploads) {
+    if (name == null || email == null) {
+      throw new IllegalArgumentException("Name and email of user requesting support is required");
+    }
+    if (subject == null) {
+      throw new IllegalArgumentException("Support ticket subject is required");
+    }
+    if (description == null) {
+      throw new IllegalArgumentException("Support ticket description is required");
+    }
+    if (type == null) {
+      throw new IllegalArgumentException("Support ticket type is required");
+    }
+    if (url == null) {
+      throw new IllegalArgumentException("Support ticket url is required");
+    }
+
+    Ticket ticket = new Ticket(
+        new Ticket.Requester(name, email),
+        subject,
+        createComment(description, url, uploads));
+    ticket.setCustomFields(createCustomFields(name, type, email, description));
+    ticket.setTicketFormId(360000669472L);
```

Comment:
What are these constants for? Could we assign them to a variable with a name or put a comment?

Reply:
I'll add some comments here, they come from the original UI implementation: https://github.com/DataBiosphere/duos-ui/blob/2b5d3f777019de681d2fb808193a1ac79fbad53e/src/libs/ajax/Support.js#L6